File: todo-frontend/src/Services/TodoService.js
import axios from "axios";

const REST_BASE_URL = "http://localhost:8080/api/todos";

export const getAllTodos = () => axios.get(REST_BASE_URL);

export const getTodoById = (todoId) => axios.get(REST_BASE_URL + "/" + todoId);

export const createTodo = (todo) => axios.post(REST_BASE_URL, todo);

export const updateTodo = (todoId, todo) =>
  axios.put(REST_BASE_URL + "/" + todoId, todo);

export const deleteTodo = (todoId) =>
  axios.delete(REST_BASE_URL + "/" + todoId);

export const completeTodo = (todoId) =>
  axios.patch(REST_BASE_URL + "/complete/" + todoId);

export const inCompleteTodo = (todoId) =>
  axios.patch(REST_BASE_URL + "/incomplete/" + todoId);

File: todo-backend/src/main/java/net/lakhwan/todo/service/TodoService.java
package net.lakhwan.todo.service;

import net.lakhwan.todo.entity.Todo;
import net.lakhwan.todo.dto.TodoDto;
import org.springframework.stereotype.Service;

import java.util.List;

public interface TodoService {
    TodoDto createTodo(TodoDto todoDto);

    TodoDto getTodoById(Long todoId);

    List<TodoDto> getAllTodos();

    TodoDto updateTodo(Long todoId, TodoDto updatedTodoDto);

    void deleteTodo(Long todoId);

    TodoDto completeTodo(Long todoId);

    TodoDto inCompleteTodo(Long todoId);
}
